src/middlewared/middlewared/plugins/kubernetes_linux/lifecycle.py

```diff
@@ -376,10 +414,27 @@ def kubernetes_dataset_custom_props(self, ds: str) -> Dict:
 
     @private
     async def start_service(self):
+        await self.before_start_check()
         await self.middleware.call('k8s.migration.scale_version_check')
         await self.middleware.call('k8s.migration.run')
         await self.middleware.call('service.start', 'kubernetes')
 
+    @private
+    async def before_start_check(self):
+        try:
+            await self.middleware.call('kubernetes.validate_k8s_fs_setup')
+        except CallError as e:
+            if e.errno != CallError.EDATASETISLOCKED:
+                await self.middleware.call(
+                    'alert.oneshot_create',
+                    'ApplicationsConfigurationFailed',
+                    {'error': e.errmsg},
+                )
+            else:
```

Comment:
This `else:` branch should be moved out of the exception scope right? Otherwise, an alert will be generated and potentially never removed? i.e.
```
try:
    blah
 except:
     create alert
 else:
     delete alert

Reply:
This alert is dismissed in another place as well when k8s starts so we know we are good but for this specific case we should have it outside the except scope. Thanks for pointing it out!